docs/book/src/plugins/extending/custom-markers.md

```diff
@@ -0,0 +1,349 @@
+                Universe: req.Universe,
+                Files: []machinery.File{mainFile},
+            }
+        },
+    }
+
+    external.Run(p)
+}
+```
+
+## Additional Examples
+
+### Java Markers
+
+```go
+type JavaMarker struct {
+    prefix  string
+    comment string
+    value   string
+}
+
+func NewJavaMarker(path string, value string) (JavaMarker, error) {
+    ext := filepath.Ext(path)
+    if ext != ".java" {
+        return JavaMarker{}, fmt.Errorf("expected .java file, got %s", ext)
+    }
+
+    return JavaMarker{
+        prefix:  "+java:scaffold:",
+        comment: "//",
+        value:   value,
+    }, nil
+}
+```
+
+### Template File Markers
+
+```go
+type TemplateMarker struct {
+    prefix  string
+    value   string
+}
+
+func NewTemplateMarker(path string, value string) (TemplateMarker, error) {
+    ext := filepath.Ext(path)
+    if ext != ".tpl" && ext != ".tmpl" {
+        return TemplateMarker{}, fmt.Errorf("expected template file, got %s", ext)
+    }
+
+    return TemplateMarker{
+        prefix: "+template:scaffold:",
+        value:  value,
+    }, nil
+}
+
+func (m TemplateMarker) String() string {
+    return fmt.Sprintf("{{/* %s%s */}}", m.prefix, m.value)
+}
+```
+
+## Best Practices
+
+1. **Use Clear Prefixes**: Choose a unique prefix for your plugin to avoid conflicts (e.g., `+rust:scaffold:`, `+java:scaffold:`)
+
+2. **Handle Comments Correctly**: Different languages have different comment syntax. Make sure to map the correct comment style for each file extension.
+
+3. **Error Handling**: Always validate file extensions and return clear error messages when unsupported files are encountered.
+
+4. **Maintain Compatibility**: When possible, follow the same patterns as Kubebuilder's core marker system to maintain consistency.
+
+5. **Document Your Markers**: Clearly document what markers your plugin supports and where they should be placed.
+
+6. **Testing**: Test your marker implementation with various file types and edge cases.
+
+## Integration with Scaffolding
+
+When using custom markers in your scaffolding logic:
+
+```go
+func (s *Scaffolder) Execute() error {
+    // Read existing file
+    content, err := afero.ReadFile(s.fs, "src/main.rs")
+    if err != nil {
+        return err
+    }
+
+    // Create marker
+    marker, err := markers.NewRustMarker("src/main.rs", "imports")
+    if err != nil {
+        return err
+    }
+
+    // Find marker in content
+    lines := strings.Split(string(content), "\n")
+    for i, line := range lines {
+        if strings.Contains(line, marker.String()) {
+            // Insert new code after marker
+            newCode := "use my_crate::MyStruct;"
+            lines = append(lines[:i+1],
+                append([]string{newCode}, lines[i+1:]...)...)
+            break
+        }
+    }
+
+    // Write back to file
+    return afero.WriteFile(s.fs, "src/main.rs",
+        []byte(strings.Join(lines, "\n")), 0644)
+}
+```
+
+## Conclusion
```

Comment:
Could you please review the other documentation we have? 
We do not use a conclusion. We should always follow the same layout and standards.